plugins/scaffolder-backend-module-confluence-to-markdown/src/actions/confluence/helpers.ts

```diff
@@ -51,10 +51,13 @@ export const readFileAsString = async (fileDir: string) => {
 export const fetchConfluence = async (relativeUrl: string, config: Config) => {
   const baseUrl = config.getString('confluence.baseUrl');
   const token = config.getString('confluence.token');
-  const response: Response = await fetch(`${baseUrl}${relativeUrl}`, {
+  const isCloud = config.getOptionalBoolean('confluence.isCloud') || false;
+  const authToken = isCloud ? `Basic ${token}` : `Bearer ${token}`;
```

Comment:
this whole dance seems to be suitable to break out into a separate function like

```ts
function parseConfig(config: Config): { authHeader: string; baseUrl: string }
```

or so, and use that below too

Reply:
Yes, this got refactored into something I feel is much nicer 🚀